akka-stream-typed/src/main/scala/akka/stream/typed/javadsl/ActorFlow.scala

```diff
@@ -134,4 +135,48 @@ object ActorFlow {
       .askWithStatus[I, Q, A](parallelism)(ref)((i, ref) => makeMessage(i, ref))(timeout.toMillis.millis)
       .asJava
 
+  /**
+   * Use the `ask` pattern to send a request-reply message to the target `ref` actor without including the context.
+   */
+  def askWithContext[I, Q, A, Ctx](
+      ref: ActorRef[Q],
+      timeout: java.time.Duration,
+      executionContextExecutor: ExecutionContextExecutor,
+      makeMessage: BiFunction[I, ActorRef[A], Q]): Flow[(I, Ctx), (A, Ctx), NotUsed] =
+    akka.stream.typed.scaladsl.ActorFlow
+      .askWithContext[I, Q, A, Ctx](parallelism = 2)(ref)((i, ref) => makeMessage(i, ref))(
+        JavaDurationConverters.asFiniteDuration(timeout),
+        executionContextExecutor)
+      .asJava
+
+  /**
+   * Use for messages whose response is known to be a [[akka.pattern.StatusReply]]. When a [[akka.pattern.StatusReply#success]] response
+   * arrives the future is completed with the wrapped value, if a [[akka.pattern.StatusReply#error]] arrives the future is instead
+   * failed.
+   */
+  def askWithStatusAndContext[I, Q, A, Ctx](
+      ref: ActorRef[Q],
+      timeout: java.time.Duration,
+      executionContextExecutor: ExecutionContextExecutor,
+      makeMessage: BiFunction[I, ActorRef[StatusReply[A]], Q]): Flow[(I, Ctx), (A, Ctx), NotUsed] =
+    akka.stream.typed.scaladsl.ActorFlow
+      .askWithStatusAndContext[I, Q, A, Ctx](parallelism = 2)(ref)((i, ref) => makeMessage(i, ref))(
+        JavaDurationConverters.asFiniteDuration(timeout),
+        executionContextExecutor)
+      .asJava
+
+  /**
+   * Use the `ask` pattern to send a request-reply message to the target `ref` actor without including the context.
+   */
+  def askWithContext[I, Q, A, Ctx](
+      parallelism: Int,
+      ref: ActorRef[Q],
+      timeout: java.time.Duration,
+      executionContextExecutor: ExecutionContextExecutor,
+      makeMessage: (I, ActorRef[A]) => Q): Flow[(I, Ctx), (A, Ctx), NotUsed] =
```

Comment:
I honestly have no idea about it... I literally copied the signature of the base `ask` method.

My reasoning was: if it works for that, it'll work for my variant too :man_shrugging:

Reply:
A quick check shows that we get `scala.Function1` which is a SAM, but does not extend `java.util.Function` on Scala 2.13 at least, so the existing signature is incorrect.

You could still call it from Java as long as you define the lambda inline, but if someone tried to pass a Java `Function` from elsewhere that wouldn't work which is not nice ergonomics for Java devs. 

You don't need to fix the existing one, but make these new factories take the correct Java function thanks.

Reply:
Ok I've changed the signature, now it takes a `BiFunction`.

I did not touch the base `ask` method since I'm not sure that's safe to do. In case it is and you think it is worth changing it let me know and I'll do it.

Reply:
That's good, let's keep it as is (and someone can possibly revisit that separately if they feel it is important enough)